### PR TITLE
4777 datafile suggested citation with file doi

### PR DESCRIPTION
**Pull request by qqmyers**
## New Contributors

Welcome! New contributors should at least glance at [CONTRIBUTING.md](/CONTRIBUTING.md), especially the section on pull requests where we encourage you to reach out to other developers before you start coding. Also, please note that we measure code coverage and prefer you write unit tests. Pull requests can still be reviewed without tests or completion of the checklist outlined below. Thanks!

## Related Issues

- connects to #4777 : Datafile suggested citation doesn't include file DOI

## Pull Request Checklist

- [ ] Unit [tests][] completed
- [ ] Integration [tests][]: None
- [ ] Deployment requirements, [SQL updates][], [Solr updates][], etc.: None
- [ ] [Documentation][docs] completed
- [ ] Merged latest from "develop" [branch][] and resolved conflicts

[tests]: http://guides.dataverse.org/en/latest/developers/testing.html
[SQL updates]: https://github.com/IQSS/dataverse/tree/develop/scripts/database/upgrades
[Solr updates]: https://github.com/IQSS/dataverse/blob/develop/conf/solr/7.3.0/schema.xml
[docs]: http://guides.dataverse.org/en/latest/developers/documentation.html
[branch]: http://guides.dataverse.org/en/latest/developers/branching-strategy.html

### CODE REVIEW

**Comment by qqmyers** [comment]
I consider this PR to be in-progress, awaiting further discussion of the on-screen citation display and how to handle the exportable citation formats. 
The citation display has been updated to be in-sync with the community forum discussion, adding a new 'direct' file citation option that uses the file's DOI, rather than using the dataset DOI followed by the file name. The PR shows both options and has Bundle text strings to explain them.

The more significant code changes are related to my refactoring to bring the various export formats into one class and to run everything through the DataCitation class. The existing code had slight variations in how, for example, harvested datasets were handled, and whether the configured citation date was used or if the publication date was hardcoded. My sense was just that the code had been written over time, so I tried to use the 'best of' option in the DataCitation class and was able to consolidate the retrieval/parsing of info for the exported citations into one class with two constructors - one for datasets starting with a DatasetVersion, and the other for files starting from FileMetadata. I added one parameter - boolean direct - that, if true for file citations, retrieves the file DOI rather than the dataset one.

With these changes, I was able to bring the code for exporting Bibtex, RIS, and Endnote into the class, and reconfigured them to directly write to a steam (which is what's really needed in FileDownloadServiceBean). 

I expect further discussion might adjust the citation display and probably will change the specifics of what's written in the bibtex, RIS, and Endnote outputs, but I wanted to share the code and perhaps get opinions on the general refactoring - if there's some reason not to consolidate the parsing code, or there's a better approach, I can probably make that type of change in parallel with ongoing discussion of the specifics of the display and file contents.

**Comment by adam3smith** [comment]
Thinking more about this, I'm not convinced what we're doing with this is optimal. As of now we have three different citations:
1. Dataset citation
2. Dataset with file citation (equivalent of current data file citation)
3. Data file citation.

1. is on the dataset landing page, 2 and 3 are on the data file landing page.

Logically these also correspond to three different sets of metadata. I think that's very confusing. With two different versions of a file citation on the file landing page, it's also unclear which of them should be encoded in schema.org metadata on that page.

I would suggest, instead, to make 1. and 2. identical, so we have 
1. Dataset citation 
2. Data file citation

In other words, get rid of the <filename>[file name] at the end of the dataset citation on file landing pages.

If people want to cite a specific file, they should just use 2. In the more common case, where they're looking at a file but really mean to cite the whole dataset, we should have 1. available on the file landing page (as per the discussion in the community group). Thoughts?

**Comment by adam3smith** [comment]
To illustrate: Currently implemented
image

Suggested change:
image

(recall that QDR uses "data project" instead of "dataset")

**Comment by scolapasta** [comment]
@adam3smith I had been wondering the same thing. Would you be ok to copy these comments into the issue itself?

**Comment by adam3smith** [comment]
Sorry, not clear which issue itself you're referring to?

**Comment by scolapasta** [comment]
#4777 
(you put the comments int the PR, rather than the issue)

**Comment by adam3smith** [comment]
oops, sorry; this was meant for 4777 in the first place

**Comment by qqmyers** [comment]
I think this PR may be ready again for review. It should now contain @adam3smith 's updates to the RIS and endnote exports and the consensus to provide a dataset and file citations on the file page (with the latter using the file DOI if published and appending the filename to the dataset citation if not).

File: src/main/webapp/file.xhtml [hunk 1]
```diff
@@ -222,51 +223,95 @@
...
                                         </div>
+                                    </div>
+                                </div>
+                            </div>
+                            <div class="col-xs-12">
+                                <div class="alert alert-warning" style="margin:0;padding:0 .5em;background:none;border-radius:0;border:0;color:#333;">
+                                  <h:outputFormat value="#{bundle['file.citation.datafile']}" escape="false" />
+                                </div>
+                            </div>
+                            <div id="citation-block-2" class="col-xs-12">
```
Comment (mheppler) [inline src/main/webapp/file.xhtml]: While re-using the ID and adding a "-2" to the end solves the duplicate ID warnings when validating, it breaks the CSS being applied to the original ID. I am working on a commit to resolve these minor layout issues, and maybe refactor some of the original CSS code to make it more scalable.

File: src/main/webapp/file.xhtml [hunk 2]
```diff
@@ -109,6 +109,7 @@
                                         </button>
                                         <ul class="dropdown-menu pull-right text-left" role="menu">
                                             <ui:fragment rendered="#{systemConfig.provCollectionEnabled}">
+                                            <!--  Should be FilePage.isLockedFromEdits() ? -->
```
Comment (mheppler) [inline src/main/webapp/file.xhtml]: Good question. Is this a question? I will look into this more.

File: src/main/java/Bundle.properties [hunk 3]
```diff
@@ -1648,8 +1648,11 @@ file.share.fileShare.tip=Share this file on your favorite social media networks.
 file.share.fileShare.shareText=View this file.
 file.title.label=Title
 file.citation.label=Citation
-file.citation.notice=This file is part of "{0}". If you use this file, please cite the dataset:
+file.citation.notice=This file is part of "{0}".
```
Comment (mheppler) [inline src/main/java/Bundle.properties]: What originally was one message, on one line, is now two messages, on two lines. It unnecessarily takes up more real estate. I'm going to try to clean this up.

File: src/main/webapp/file.xhtml (hunk 2 above)
Comment (qqmyers) [reply src/main/webapp/file.xhtml]: ~yes. Seems like a cut/paste error but separate from what I was working on.

**Comment by pdurbin** [comment]
#4971 is the new version of the pull request so I'm closing this one.
